### PR TITLE
chore: retire Makefile in favor of yarn

### PR DESCRIPTION
**Pull request by starsirius**
The type of this PR is: **Chore**

This PR resolves [CX-1895]

### Description

https://artsy.slack.com/archives/C02BAQ5K7/p1630615328076200

The current [Makefile](https://github.com/artsy/eigen/blob/ce56fa147cac4953e291b5d1517b637bf3859772/Makefile) is just to run individual scripts. We're using Yarn scripts for similar purposes (e.g. [`make help`](https://github.com/artsy/eigen/blob/ce56fa147cac4953e291b5d1517b637bf3859772/Makefile#L20-L21) and [`yarn help`](https://github.com/artsy/eigen/blob/ce56fa147cac4953e291b5d1517b637bf3859772/package.json#L30)), or directly running the scripts (e.g. [on CI](https://github.com/artsy/eigen/blob/ce56fa147cac4953e291b5d1517b637bf3859772/.circleci/config.yml#L69)).

Until we need more sophisticated build tools from `make`, let's retire it. I think we can either directly run the scripts or add Yarn scripts, if helpful.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Retire Makefile in favor of Yarn scripts

</details>

[CX-1895]: https://artsyproduct.atlassian.net/browse/CX-1895

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Retire Makefile in favor of Yarn scripts)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 52c08c2ee2d412cbea69e4cc3b99579c31a75453
</p>

File: package.json [hunk 1]
```diff
@@ -27,7 +27,10 @@
     "clean-ios": "yarn rimraf ~/Library/Developer/Xcode/DerivedData",
     "clean-android": "cd android; ./gradlew clean cleanBuildCache; cd -",
     "clean": "yarn clean-ios; yarn clean-android",
-    "help": "scripts/help.js",
+    "flip_table": "./scripts/flip-table",
+    "flip_table_extreme": "./scripts/flip-table-extreme",
```
Comment (starsirius) [inline package.json:31]: I ported over the flip table scripts here that seems to be useful. Is there any others I should add too?

File: package.json [hunk 2]
```diff
@@ -27,7 +27,10 @@
...
+    "help": "./scripts/help.js",
+    "install:all": "./scripts/install",
```
Comment (starsirius) [inline package.json:33]: Added this `yarn install:all` from [`make install`](https://github.com/artsy/eigen/blob/ce56fa147cac4953e291b5d1517b637bf3859772/Makefile#L14-L15), not to be confused with `yarn install`.

File: package.json [hunk 3]
```diff
@@ -37,6 +40,8 @@
     "prettier-project": "yarn run prettier-write 'src/**/*.{ts,tsx}'",
     "prettier-write": "prettier --write",
     "relay": "relay-compiler",
+    "setup:artsy": "./scripts/setup-env-for-artsy",
+    "setup:oss": "./scripts/setup-env-for-oss",
```
Comment (starsirius) [inline package.json:44]: Also added the setup scripts that seem to be used more often.

File: package.json [hunk 4]
```diff
@@ -27,7 +27,10 @@
     "clean-ios": "yarn rimraf ~/Library/Developer/Xcode/DerivedData",
     "clean-android": "cd android; ./gradlew clean cleanBuildCache; cd -",
     "clean": "yarn clean-ios; yarn clean-android",
-    "help": "scripts/help.js",
```
Comment (starsirius) [inline package.json:30]: Lol. Realized `yarn help` is going to run the help manual for `yarn` command itself. Looks like we need a new name.

Comment (starsirius) [reply package.json:30]: Renamed it to `yarn doctor` in https://github.com/artsy/eigen/pull/5393/commits/3cd59bccbf56ae846047475e3983b416ad9ef126 (borrowed the name from `brew doctor`). See if you like it!

File: package.json (hunk 1 above)
Comment (MounirDhahri) [reply package.json:32]: no these two should be enough for us

File: package.json (hunk 2 above)
Comment (MounirDhahri) [reply package.json:33]: ah this is awesome

File: package.json (hunk 4 above)
Comment (MounirDhahri) [reply package.json:30]: that makes sense for me. it's also common in react-native as in here https://reactnative.dev/blog/2019/11/18/react-native-doctor

**Review by MounirDhahri — APPROVED** [review]
Great one 💜 Thank you @starsirius

File: Makefile [hunk 5]
```diff
@@ -1,56 +0,0 @@
...
-	./scripts/deploy-both
-
-deploy:
-	./scripts/deploy-both
-
-deploy_ios:
-	./scripts/deploy-ios
-
-deploy_android:
-	./scripts/deploy-android
```
Comment (MounirDhahri) [inline Makefile:36]: **note:** it makes sense to not replace this since our documentation references to `./scripts/deploy-*` instead of `make deploy-*`

**Review by brainbicycle — APPROVED** [review]
Nice one! Thanks for following up on this!

**Pull request merged**